src/context/useReducer.js

```diff
@@ -0,0 +1,115 @@
+export const data = {
+  cartItems: [],
+  wishList: [],
+  filteredData: []
+};
+
+const SET_NEW_DATA = "setNewData";
+const SET_CART_ITEMS = "setCartItems";
+const ADD_CART_ITEM = "addCartItem";
+const INC_QTY = "incQty";
+const DEC_QTY = "decQty";
+const REMOVE_CART_ITEM = "removeCartItem";
+const ADD_WISHLIST_ITEM = "addWishlistItem";
+const REMOVE_WISHLIST_ITEM = "removeWishlistItem";
+const SET_WISHLIST_ITEMS = "setWishlistItems";
+const HIGH_TO_LOW = "highToLow";
+const LOW_TO_HIGH = "lowToHigh";
+const ADD_CART_ITEM_AND_REMOVE_FROM_WISHLIST = "addCartItem&removeWishlistItem";

```

Comment:
`ADD_CART_ITEM_AND_REMOVE_FROM_WISHLIST ` this dispatch type shouldnt be needed imo Rahul. Move to Cart btn can use our existing functions